### PR TITLE
issue 794 marketing campaign email

### PR DESCRIPTION
**Pull request by kylewu**
https://github.com/gitcoinco/web/issues/794

### CODE REVIEW

**Comment by kylewu** [comment]
I've not added GIF files in email.
But could you take a look at my approach to see if it's acceptable.
I'll do some updates after your feedback

**Comment by kylewu** [comment]
@PixelantDesign @vs77bb  could you take a look at this pr?

**Comment by owocki** [comment]
checking this out now

File: app/marketing/management/commands/campaign_email.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        created_on__lte=n_days_ago(nth-1))
+    for sub in subs:
+        send_nth_email_to_subscriber(nth, sub)
+
+
+class Command(BaseCommand):
+
+    help = 'Send marketing email to new subscribers'
+
+    def handle(self, *args, **options):
```
Comment (owocki) [inline app/marketing/management/commands/campaign_email.py]: can you schedule this crontab file?

File: app/marketing/management/commands/campaign_email.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        # day 1 email has been sent
+        # start from day 2
+        for i in range(2, 7):
```
Comment (owocki) [inline app/marketing/management/commands/campaign_email.py]:
```
>>> list(range(2, 7))
[2, 3, 4, 5, 6]
```

i think this range is too narrow, no?

File: app/marketing/signals.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from marketing.mails import nth_day_email_campaign
+from marketing.models import EmailSubscriber
+
+
+@receiver(post_save, sender=EmailSubscriber)
+def create_email_subscriber(sender, instance, created, **kwargs):
```
Comment (owocki) [inline app/marketing/signals.py]: you can use `get_or_save_email_subscriber` for this too!

File: app/marketing/mails.py [hunk 4]
```diff
@@ -416,3 +416,20 @@ def setup_lang(to_email):
...
+    firstname = subscriber.email.split('@')[0]
+
+    if subscriber.profile and subscriber.profile.user and subscriber.profile.user.first_name:
+        firstname = subscriber.profile.user.first_name
+
+    cur_language = translation.get_language()
+    try:
+        setup_lang(subscriber.email)
+        from_email = settings.CONTACT_EMAIL
+        html, text, subject = render_nth_day_email_campaign(nth, firstname)
```
Comment (owocki) [inline app/marketing/mails.py]: can you wrap this in a `should_suppress_notification_email()` call

File: app/marketing/views.py [hunk 5]
```diff
@@ -738,3 +739,40 @@ def leaderboard(request, key=''):
         'podium_items': items[:3] if items else []
     }
     return TemplateResponse(request, 'leaderboard.html', context)
+
+
+@staff_member_required
+def day_1_email_campaign(request):
+    response_html, _, _ = render_nth_day_email_campaign(1, "staff member")
```
Comment (mbeacom) [inline app/marketing/views.py]: Use `day` in place of `1`

File: app/marketing/views.py [hunk 6]
```diff
@@ -738,3 +739,40 @@ def leaderboard(request, key=''):
         'podium_items': items[:3] if items else []
     }
     return TemplateResponse(request, 'leaderboard.html', context)
+
+
+@staff_member_required
+def day_1_email_campaign(request):
```
Comment (mbeacom) [inline app/marketing/views.py]: Can we modify this to take `day` ?

File: app/marketing/management/commands/campaign_email.py (hunk 1 above)
Comment (kylewu) [reply app/marketing/management/commands/campaign_email.py]: I simply run this command once and it looks ok.
There are other commands from this repo, you need to schedule this one just like the others

File: app/marketing/management/commands/campaign_email.py (hunk 2 above)
Comment (kylewu) [reply app/marketing/management/commands/campaign_email.py]: we send 6 emails to one subscriber, here we control 5, and the first one is in signals

File: app/app/urls.py [hunk 7]
```diff
@@ -193,6 +193,12 @@
     url(r'^_administration/email/start_work_expire_warning$', retail.emails.start_work_expire_warning, name='start_work_expire_warning'),
     url(r'^_administration/email/start_work_expired$', retail.emails.start_work_expired, name='start_work_expired'),
     url(r'^_administration/email/new_tip/resend$', retail.emails.resend_new_tip, name='resend_new_tip'),
+    url(r'^_administration/email/day_1_email_campaign$', marketing.views.day_1_email_campaign, name='day_1_email_campaign'),
+    url(r'^_administration/email/day_2_email_campaign$', marketing.views.day_2_email_campaign, name='day_2_email_campaign'),
+    url(r'^_administration/email/day_3_email_campaign$', marketing.views.day_3_email_campaign, name='day_3_email_campaign'),
+    url(r'^_administration/email/day_4_email_campaign$', marketing.views.day_4_email_campaign, name='day_4_email_campaign'),
+    url(r'^_administration/email/day_5_email_campaign$', marketing.views.day_5_email_campaign, name='day_5_email_campaign'),
+    url(r'^_administration/email/day_6_email_campaign$', marketing.views.day_6_email_campaign, name='day_6_email_campaign'),
```
Comment (mbeacom) [inline app/app/urls.py]: Can we change this to something like:  `path('_administration/email/day_email_campaign/<int:day>', marketing.views.day_email_campaign, name='day_email_campaign'),`

Comment (kylewu) [reply app/app/urls.py]: good one! i'll update

File: app/marketing/mails.py (hunk 4 above)
Comment (kylewu) [reply app/marketing/mails.py]: do you mean this

File: app/app/urls.py (hunk 7 above)
Comment (kylewu) [reply app/app/urls.py]: updated in f9f865f

File: app/retail/templates/emails/email_campaign_day_2.txt [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+Hey {{firstname}},
+
+By this point, you might have poked around the Gitcoin Issue Explorer (https://gitcoin.co/explorer?q=) to see the open source projects you could make money by contributing! If interested in doing work, here’s how:
+
+Start Work
+Click ‘Start Work’ to kick off a message to the repo-maintainer that you’re interested in the issue
+Respond on the Github thread with your intended approach for the bounty
+
+[GIF of Start Work]
```
Comment (owocki) [inline app/retail/templates/emails/email_campaign_day_2.txt]: i think you can remove these tags from the txt version

Comment (kylewu) [reply app/retail/templates/emails/email_campaign_day_2.txt]: right! fixed in e4c7ca3

File: app/marketing/management/commands/campaign_email.py (hunk 1 above)
Comment (owocki) [reply app/marketing/management/commands/campaign_email.py]: better yet.. you could just update the crontab on this branch https://github.com/gitcoinco/web/blob/master/scripts/crontab

Comment (kylewu) [reply app/marketing/management/commands/campaign_email.py]: @owocki lol, i searched cron in text, but not filename 😳 
fixed in 33de5af

**Comment by owocki** [comment]
testing now

File: app/marketing/management/commands/campaign_email.py (hunk 2 above)
Comment (owocki) [reply app/marketing/management/commands/campaign_email.py]: thanks

Comment (owocki) [reply app/marketing/management/commands/campaign_email.py]: thanks

File: app/marketing/mails.py (hunk 4 above)
Comment (owocki) [reply app/marketing/mails.py]: yes that would be acceptable

**Comment by owocki** [comment]
great.. @vs77bb @PixelantDesign we good to ship this?

**Comment by kylewu** [comment]
@owocki I received one email from @PixelantDesign early this week, that the growth marketer would like to add some tracking which is to update some urls. But I have not received any update yet.

IMO it's better to merge this because there are always some conflicts and in this PR we do not enable this command in cron. It's not very bad to update url in another PR.

**Comment by PixelantDesign** [comment]
@kylewu, your approach spunds good. Thanks!

**Comment by owocki** [comment]
i just approved this to be merged once the conflicts are resolved

@PixelantDesign lets keep an eye on the numbers once this goes out.. my gut says 

1. there's too much text in these emails and we may want to create 
2. we'll may be sending too many emails by sending 1 every day (on top of the daily bounty email)
3. we'll need to update the email settings page to account for the existence of welcome mails

**Comment by owocki** [comment]
@mbeacom any comments here?

File: app/marketing/apps.py [hunk 9]
```diff
@@ -24,3 +24,6 @@
 class MarketingConfig(AppConfig):
 
     name = 'marketing'
+
+    def ready(self):
+        import marketing.signals  # noqa
```
Comment (mbeacom) [inline app/marketing/apps.py]: Can we be explicit here and define: `from .signals import create_email_subscriber` and remove the `noqa`

File: app/marketing/views.py [hunk 10]
```diff
@@ -498,3 +498,11 @@ def leaderboard(request, key=''):
         'podium_items': items[:3] if items else []
     }
     return TemplateResponse(request, 'leaderboard.html', context)
+
+
+@staff_member_required
+def day_email_campaign(request, day):
+    if day not in list(range(1, 6)):
+        raise Http404
+    response_html, _, _ = render_nth_day_email_campaign('foo@bar.com', day, "staff member")
```
Comment (mbeacom) [inline app/marketing/views.py]: You'll want to use `response_html, __, __ =` here due to redeclaration of `gettext_lazy as _`  and avoid future confusion.

File: app/retail/emails.py [hunk 11]
```diff
@@ -332,6 +333,27 @@ def render_bounty_startwork_expired(to_email, bounty, interest, time_delta_days)
...
+        5: "Day 5: The Gitcoin Community",
+    }
+
+    subject = subject_map[nth]
+
+    params = {
+        'firstname': firstname,
+        'subscriber': get_or_save_email_subscriber(to_email, 'internal'),
+    }
+    response_html = premailer_transform(render_to_string("emails/email_campaign_day_{}.html".format(nth), params))
```
Comment (mbeacom) [inline app/retail/emails.py]: Can we use fstrings here?

File: app/retail/templates/emails/email_campaign_day_4.html [hunk 12]
```diff
@@ -0,0 +1,74 @@
+{% extends 'emails/template.html' %} {% comment %} Copyright (C) 2018 Gitcoin Core This program is free software: you can
+redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version. This program is distributed in the hope
+that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of the GNU Affero General
+Public License along with this program. If not, see
+<http://www.gnu.org/licenses/>. {% endcomment %} {% load i18n humanize %} {% block content %}
```
Comment (mbeacom) [inline app/retail/templates/emails/email_campaign_day_4.html]: Can we fix the formatting here?

File: app/retail/templates/emails/email_campaign_day_2.html [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+    {% blocktrans %}
+    By this point, you might have poked around the Gitcoin <a href="https://gitcoin.co/explorer?q=">Issue Explorer</a> to
+    see the open source projects you could make money by contributing!
+    If interested in doing work, here's how:
+    {% endblocktrans %}
+    </p>
+
+    <h2>Start Work</h2>
+    <p>
+        {% blocktrans %}
```
Comment (mbeacom) [inline app/retail/templates/emails/email_campaign_day_2.html]: Can we fix all of these .html files to follow 2-space indentation?

**Comment by kylewu** [comment]
@mbeacom all fixed in 821c9dd

File: app/marketing/apps.py [hunk 14]
```diff
@@ -24,3 +24,6 @@
 class MarketingConfig(AppConfig):
 
     name = 'marketing'
+
+    def ready(self):
+        from .signals import create_email_subscriber
```
[the remark this answers is not in the document]
Comment (kylewu) [reply app/marketing/apps.py]: @mbeacom ^^ looks like this fails

**Comment by kylewu** [comment]
@mbeacom @owocki could you review the code again? I've fixed some code based on @mbeacom comments last week

**Comment by mbeacom** [comment]
@kylewu Sure.  Would you like us to fix the merge conflicts or are you planning on tackling it?

**Comment by kylewu** [comment]
@mbeacom merged lasted master in 430899a

**Comment by kylewu** [comment]
@owocki would you mind checking the status of this PR? Is it ok to merge?

**Comment by kylewu** [comment]
@owocki fixed in d5c2ac9

**Comment by owocki** [comment]
@PixelantDesign @vs77bb i move we merge this and keep an 🦅 👁 on the email unsubscribe numbers next week.  that sound okay to you?

**Comment by PixelantDesign** [comment]
sounds good!
cc: @mbeacom

**Comment by kylewu** [comment]
@owocki any more comments on this?

**Comment by vs77bb** [comment]
@mbeacom Are we kicking off these marketing e-mails now? I can take a close look at the copy, if so. I think it's a good idea to do so!

**Comment by mbeacom** [comment]
:star: PR built and available in a preview environment **gitcoinco-web-pr-1010** [here](http://web.jx-gitcoinco-web-pr-1010.ci.gitcoin.co)

**Comment by PixelantDesign** [comment]
we may need to look at all of the screenshots, it's likely they're outdated. 

cc: @willsputra

**Comment by vs77bb** [comment]
@owocki @thelostone-mc What are the chances we can merge this in the next merge day, using the updated [copy found here](https://docs.google.com/document/d/1eO2aWj1RhbjKzOfBQWVQxyQ_0jamTRkr7jjoFVjEIdU/edit) for the drip campaigns?

**Comment by thelostone-mc** [comment]
closing this out in favor of #3766

**Pull request closed without merging**